integration-tests/test/wait.py

```diff
@@ -65,6 +65,18 @@ def is_satisfied(self) -> bool:
         peers = int(match[1])
         return peers >= self.minimum_peers_number
 
+class NodeSeesBlock:
+    def __init__(self, node: 'Node', block_hash: str) -> None:
+        self.node = node
+        self.block_hash = block_hash
+
+    def __str__(self) -> str:
+        args = ', '.join(repr(a) for a in (self.node.name, self.block_hash))
+        return '<{}({})>'.format(self.__class__.__name__, args)
+
+    def is_satisfied(self) -> bool:
+        block_message = self.node.get_block(self.block_hash)
+        return "Error: Failure to find block with hash" not in block_message
```

Comment:
Hmm, I think `get_block()` should be returning non-zero exit code if something goes wrong and non-zero exit codes translate into a Python exception.